### PR TITLE
Typed stream adapters, #23604

### PR DESCRIPTION
**Pull request by patriknw**
Refs #23604

Cont from https://github.com/akka/akka/pull/24289
Fixed DslFactoriesConsistencySpec

### CODE REVIEW

File: akka-stream-typed/src/test/java/docs/akka/stream/typed/ActorSourceExample.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com/>
+ */
+
+package docs.akka.stream.typed;
+
+// #actor-source-ref
```
Comment (ktoso) [inline akka-stream-typed/src/test/java/docs/akka/stream/typed/ActorSourceExample.java]: weird space, will still work?

**Review by ktoso — APPROVED** [review]
LGTM, thanks for taking it over the finish line

**Comment by patriknw** [comment]
another rebase (index.md)

**Comment by patriknw** [comment]
and one more, now with mima filters

**Pull request merged**